include/circt/Dialect/FIRRTL/FIRRTLDeclarations.td

```diff
@@ -122,7 +129,7 @@ def InstanceOp : ReferableDeclOp<"instance", [HasParent<"firrtl::FModuleOp, firr
   }];
 }
 
-def MemOp : ReferableDeclOp<"mem"> {
+def MemOp : HardwareDeclOp<"mem"> {
```

Comment:
The refactoring to have a HardwareDeclOp could probably be merged directly first, it makes sense to me.